lib/config/convict.js

```diff
@@ -0,0 +1,835 @@
+  sslKeyPath: {
+    doc: 'SSL private key path absolute or relative to CodiMD\'s base director (only need when you set `useSSL`)',
+    format: 'optional-path',
+    default: ''
+  },
+  sslCertPath: {
+    doc: 'SSL certificate path absolute or relative to CodiMD\'s base director (only need when you set `useSSL`)',
+    format: 'optional-path',
+    default: ''
+  },
+  sslCAPath: {
+    doc: 'SSL CA certificate path absolute or relative to CodiMD\'s base director (only need when you set `useSSL`)',
+    format: 'optional-path',
+    default: ''
+  },
+  dhParamPath: {
+    doc: 'SSL DH parameter absolute or relative to CodiMD\'s base director (only need when you set `useSSL`)',
+    format: 'optional-path',
+    default: ''
+  },
+  // other path
+  viewPath: {
+    doc: 'Frontend template directory path.',
+    format: 'existing-path',
+    default: './public/views'
+  },
+  tmpPath: {
+    doc: 'Path to temporarily store data.',
+    format: 'existing-path',
+    default: os.tmpdir()
+  },
+  defaultNotePath: {
+    doc: 'Path to the default note content document. Can be used to create templates, prefill all notes, and provide default settings in note metadata',
+    format: 'existing-path',
+    default: './public/default.md'
+  },
+  docsPath: {
+    doc: 'Path to public docs. Contains files like the release notes, impress, slide-example, features page and more.',
+    format: 'existing-path',
+    default: './public/docs'
+  },
+  uploadsPath: {
+    doc: 'Path to uploads directory. When local uploads are allowed, the path to store pictures uploaded to the instance.',
+    format: 'existing-path',
+    default: './public/uploads'
+  },
+  // session
+  sessionName: {
+    doc: 'Name used by the cookies that stores the session information.',
+    format: 'String',
+    default: 'connect.sid'
+  },
+  sessionSecret: {
+    doc: 'Secret used to encrypt session IDs and details.',
+    format: 'String',
+    env: 'CMD_SESSION_SECRET',
+    default:'secret',
+    sensitive: true
+  },
+  sessionSecretLen: {
+    doc: 'Length of the session secret generated when no session secret is defined in config.',
+    format: 'nat',
+    default: 128
+  },
+  sessionLife: {
+    doc: 'Maximum time a session can be used without re-login.',
+    format: 'nat',
+    env: 'CMD_SESSION_LIFE',
+    default: 14 * 24 * 60 * 60 * 1000 // 14 days
+  },
+  staticCacheTime: {
+    doc: 'Time that static assets delivered from CodiMD are cached on the client.',
+    format: 'nat',
+    default: 1 * 24 * 60 * 60 * 1000, // 1 day
+  },
+  // socket.io
+  heartbeatInterval: {
+    doc: 'Time interval in milliseconds used for heartbeats that are send out by the client to check that the websocket connection is still conencted.',
+    format: 'nat',
+    default: 5000
+  },
+  heartbeatTimeout: {
+    doc: 'Time in milliseconds that is allowed to pass without heartbeat before the websocket connection is considered disconnected.',
+    format: 'nat',
+    default: 10000
```

Comment:
we could verify that it's longer than `heartbeatInterval`, just in case. Not that anyone I know ever modified any of the two.

Reply:
You mean, that's it's not shorter than heartbeatInterval? Yes and now. I don't think we can already use defaults we just configured. And even when we can, my head has a hard time to wrap around that. We might want to do this validation out of band?

Reply:
we can't be the only ones who have interdependent configuration items? anyway it's probably not really important.